libass/ass_outline.c

```diff
@@ -323,6 +323,21 @@ bool outline_transform_3d(ASS_Outline *outline, const ASS_Outline *source,
     return true;
 }
 
+int32_t outline_find_min_transformed_x(const ASS_Outline *outline,
+                                       const double m[3][3]) {
+    int32_t min_x = INT32_MAX;
+    const ASS_Vector *pt = outline->points;
+    for (size_t i = 0; i < outline->n_points; i++) {
+        double z = m[2][0] * pt[i].x + m[2][1] * pt[i].y + m[2][2];
+        double x = (m[0][0] * pt[i].x + m[0][1] * pt[i].y + m[0][2]) / FFMAX(z, 0.1);
+        if (!(fabs(x) < OUTLINE_MAX))
```

Comment:
Not sure about additional failure mode here. Maybe it's better to clamp or even return full `double` without any rounding?